blockchain/chain.go

```diff
@@ -694,6 +694,164 @@ func (b *BlockChain) CalcPastMedianTime() (time.Time, error) {
+		// this input can be spent.
+		sequenceNum := txIn.Sequence
+		relativeLock := int64(sequenceNum & RelativeLockMask)
+
+		switch {
+		// Relative time locks are disabled for this input, so we can
+		// skip any further calculation.
+		case sequenceNum&RelativeLockTimeDisabled == RelativeLockTimeDisabled:
+			continue
+		case sequenceNum&RelativeLockSeconds == RelativeLockSeconds:
+			// This input requires a relative time lock expressed
+			// in seconds before it can be spent. Therefore, we
+			// need to query for the block in which this input was
+			// included within so we can compute the past median
+			// time for the block prior to the one which included
+			// this referenced output.
+			// TODO: caching should be added to keep this speedy
+			var blockNode *blockNode
+			if err := b.db.View(func(dbTx database.Tx) error {
+				header, err := dbFetchHeaderByHeight(dbTx, inputHeight-1)
+				if err != nil {
+					return err
+				}
+				blockSha := header.BlockHash()
+				blockNode = newBlockNode(header, &blockSha, inputHeight)
+				return nil
+			}); err != nil {
+				return nil, err
+			}
+			adjustedTime, err := b.calcPastMedianTime(blockNode)
+			if err != nil {
+				return sequenceLock, err
+			}
+
+			// Time based relative time-locks as defined by BIP 68
+			// have a time granularity of RelativeLockSeconds, so
+			// we shift left by this amount to convert to the
+			// proper relative time-lock. We also subtract one from
+			// the relative lock to maintain the original lockTime
+			// semantics.
+			timeLockSeconds := (relativeLock << RelativeLockSecondsGranularity) - 1
+			timeLock := adjustedTime.Unix() + timeLockSeconds
+
+			if timeLock > sequenceLock.Seconds {
+				sequenceLock.Seconds = timeLock
+			}
+		default:
+			// The relative lock-time for this input is expressed
+			// in blocks so we calculate the relative offset from
+			// the input's height as its converted absolute
+			// lock-time. We subtract one from the relative lock in
+			// order to maintain the original lockTime semantics.
+			blockHeight := inputHeight + int32(relativeLock-1)
+			if blockHeight > sequenceLock.BlockHeight {
+				sequenceLock.BlockHeight = blockHeight
+			}
+		}
+	}
+
+	return sequenceLock, nil
+}
+
+// LockTimeToSequence converts the passed relative locktime to a sequence
+// number in accordance to BIP-68.
+// See: https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki
+//  * (Compatibility)
+func LockTimeToSequence(isSeconds bool, locktime uint32) uint32 {
+	// If we're expressing the relative lock time in blocks, then the
+	// corresponding sequence number if simply the desired input age.
+	if !isSeconds {
+		return locktime
+	}
+
+	// Set the 22nd bit which indicates the lock time is in seconds, then
+	// shift the locktime over by 9 since the time granularity is in
+	// 512-second intervals (2^9). This results in a max lock-time of
+	// 33,554,431 seconds, or 1.06 years.
```

Comment:
I believe you're correct! I've updated the comment to match the proper limits on time-based relative lock times. 

Will you be submitting a PR to update the BIP?